### PR TITLE
Show focus on `Button` when interacting with keyboard

### PR DESCRIPTION
**Pull request by justinanastos**
This uses `@react-aria/focus` under the hood to handle the focus. Also, this cleans up some internals of button.

## Release Notes

* Show focus ring around `Button` when keyboard focus is received
* Use `@react-aria/utils`'s `mergeProps` in `Button`
* Add test to `Button` for `className` merging

<details>
  <summary>📦 Published PR as canary version: <code>7.17.1-canary.244.5608.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @apollo/space-kit@7.17.1-canary.244.5608.0
  # or 
  yarn add @apollo/space-kit@7.17.1-canary.244.5608.0
  ```
</details>

### CODE REVIEW

File: src/Button/Button.stories.mdx [hunk 1]
```diff
@@ -447,7 +447,7 @@ In space constrained places or for floating action buttons (FAB), we can show an
 
 ### End Icons
 
-You can specify that a button has an additional icon at the end of the row
+You can specify that a button has an additional icon at the end of the row. This will be assumed to be a menu selector and will change the button's right padding.
```
Comment (Jephuff) [inline src/Button/Button.stories.mdx:450]: This seems a bit unexpected? I would expect icon and endIcon to behave similarly. If `endIcon` is for a specific use case maybe the name should reflect that. 
Or remove `endIcon` and add a prop like isMenuSelect that changes the padding and moves the icon to the right?

File: src/Button/Button.tsx [hunk 2]
```diff
@@ -265,304 +268,280 @@ export const Button = React.forwardRef<HTMLElement, Props>(
...
+        // If the button is disabled, then don't propogate `onClick`
+        // events.
+        if (mergedProps.disabled) return event.preventDefault();
+
+        mergedProps.onClick?.(event);
+      },
+      [mergedProps]
+    );
+
+    const focusedStyles: ObjectInterpolation<any> = {
```
[the remark this answers is not in the document]
Comment (justinanastos) [reply src/Button/Button.tsx:309]: 👍

File: src/Button/Button.tsx [hunk 3]
```diff
@@ -265,304 +268,280 @@ export const Button = React.forwardRef<HTMLElement, Props>(
...
+     */
+    const onClick = React.useCallback(
+      (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        // If the button is disabled, then don't propogate `onClick`
+        // events.
+        if (mergedProps.disabled) return event.preventDefault();
+
+        mergedProps.onClick?.(event);
+      },
+      [mergedProps]
```
[the remark this answers is not in the document]
Comment (justinanastos) [reply src/Button/Button.tsx:306]: Great catch 👍

File: src/Button/Button.stories.mdx (hunk 1 above)
Comment (justinanastos) [reply src/Button/Button.stories.mdx:450]: In retrospect I wish I'd named it something more like that; but this is already in place and now will just behave correctly 🤷‍♂️

**Review by Jephuff — APPROVED** [review]

**Comment by apollo-bot2** [comment]
:rocket: PR was released in [`v7.18.0`](https://github.com/apollographql/space-kit/releases/tag/v7.18.0) :rocket:

**Pull request merged**